addons/web/models/res_partner.py

```diff
@@ -27,12 +27,12 @@ def _build_vcard(self):
         vcard = vobject.vCard()
         # Name
         n = vcard.add('n')
-        n.value = vobject.vcard.Name(family=self.name)
+        n.value = vobject.vcard.Name(family=self.name or self.complete_name)
         if self.title:
             n.value.prefix = self.title.name
         # Formatted Name
         fn = vcard.add('fn')
-        fn.value = self.name
+        fn.value = self.name or self.complete_name
         # Address
         adr = vcard.add('adr')
         adr.value = vobject.vcard.Address(street=self.street or '', city=self.city or '', code=self.zip or '')
```

Comment:
is "photo.value = b64decode(self.avatar_512)" always ok while we are here ?

Reply:
Zboing @tde-banana-odoo :wave: 

Thanks for reviewing the PR.

Yes, the value of  `self.avatar_512` is always available.